lib/node_modules/@stdlib/random/base/mt19937/lib/factory.js

```diff
@@ -62,19 +62,32 @@ var MATRIX_A = 0x9908b0df;
 * // returns 822569775
 */
 function factory( seed ) {
+	var seedIsArray = false;
 	var state;
+	var mag01 = [ 0x0, MATRIX_A ];
 	var mti;
 	var mt = new Array( N );
 	var s;
+	var i;
+	var j;
+	var k;
 
 	if ( arguments.length ) {
-		if ( !isPositiveInteger( seed ) ) {
-			throw new TypeError( 'invalid input argument. Must provide a positive integer. Value: `' + seed + '`.' );
-		}
-		if ( seed > MAX_SEED ) {
-			throw new RangeError( 'invalid input argument. Must provide a positive integer less than the maximum signed 32-bit integer. Value: `' + seed + '`.' );
+		if ( Array.isArray( seed ) ) {
+			seedIsArray = true;
+			i = 1;
+			j = 0;
+			k = ( N > seed.length ) ? N : seed.length;
+			state = 19650218 | 0; // asm type annotation
+		} else {
+			if ( !isPositiveInteger( seed ) ) {
+				throw new TypeError( 'invalid argument. Must provide a positive integer. Value: `' + seed + '`.' );
+			}
+			if ( seed > MAX_SEED ) {
+				throw new RangeError( 'invalid argument. Must provide a positive integer less than the maximum signed 32-bit integer. Value: `' + seed + '`.' );
+			}
+			state = seed | 0; // asm type annotation
 		}
-		state = seed | 0; // asm type annotation
 	} else {
 		state = randint32() | 0; // asm type annotation
```

Comment:
While we generated a "signed" 32-bit integer for LCG, is that necessary here? Can we not do `randuint32`? If so, the `asm.js` type annotations would need to be updated. In short, is the seed allowed to exceed `INT32_MAX`, but must be less than or equal to `UINT32_MAX`? If so, we should update the seeding procedure.

Reply:
Yes, the seed can be more than `INT32_MAX` and less than or equal to `UINT32_MAX`. So we can do `randuint32`. But does using a `uint32` random seed over `int32` random seed make much of a difference? The original C implementation uses a hardcoded seed (`init_genrand(5489UL); /* a default initial seed is used */`) if not given one.

Also, I never got around the `// asm type annotation` lines. I just followed the pattern. What exactly is happening there?

Reply:
IMO, seeing `randint32` makes me believe it is "intentional" (as if the algo must use `int32`), rather than "accidental". I think there is some mismatch based on the fact that the seed "array", as defined in the source, is `uint32` (unsigned long) and then an `int32` seed is used here. As for the hardcoded seed, I would avoid this and keep random generation of a seed in order to avoid automatic deterministic output, thus requiring "opt-in", rather than "opt-out" for repeated sequences. Personally, I would just amend to `randuint32`, but am open to counterarguments.

Re: asm type annotations. These come from [`asm.js`](http://asmjs.org/spec/latest/#parameter-type-annotations), the precursor to WebAssembly, which allow us to give the compiler hints as to numeric values. While JavaScript officially only has numeric values of type `double`, in practice, JavaScript has both signed and unsigned integers, but we need to indicate them as such to the compiler. So anytime we use `|0` or `>>>0` for indicating numeric types, we always add a `// asm type annotation` comment so future devs understand why the, on the surface, no-op bit operations are included in the source. They are there to offer the compiler hints so that compilers can perform various performance optimizations (e.g., guaranteed integer opts).

Reply:
Okay, so basically all the `|0` need to be changed to `>>>0`?
~~~~
function randuint32() {
	var v = floor( 1.0 + (MAX*Math.random()) ); // eslint-disable-line stdlib/no-builtin-math
	return v>>>0; // asm type annotation
}
~~~~

Reply:
Yes, wherever `uint32` applies.